core/src/avm2/globals/flash/text/font.rs

```diff
@@ -129,18 +82,66 @@ pub fn has_glyphs<'gc>(
 pub fn enumerate_fonts<'gc>(
     activation: &mut Activation<'_, 'gc>,
     _this: Object<'gc>,
-    _args: &[Value<'gc>],
+    args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    avm2_stub_method!(activation, "flash.text.Font", "enumerateFonts");
-    Ok(ArrayObject::from_storage(activation, ArrayStorage::new(0))?.into())
+    let mut storage = ArrayStorage::new(0);
+    let font_class = activation.avm2().classes().font;
+
+    if args.get_bool(0) {
+        // We could include the ones we know about, but what to do for the ones that weren't eagerly loaded?
+        avm2_stub_method!(
+            activation,
+            "flash.text.Font",
+            "enumerateFonts",
+            "with device fonts"
+        );
+    }
+
+    for font in activation.context.library.global_fonts() {
+        storage.push(FontObject::for_font(activation.context.gc_context, font_class, font).into());
+    }
+
+    if let Some(movie) = activation.caller_movie() {
```

Comment:
This should always be `Some`- if it isn't, we have a bug. With #13492 it should always be set except in the strangest of cases (no SWF should be calling those cases).